packages/types/src/hooks.ts

```diff
@@ -441,11 +451,11 @@ export type OnSubscribeEventPayload<ContextType> = {
 /**
  * Payload with which the onSubscribeResult hook is invoked.
  */
-export type OnSubscribeResultEventPayload<ContextType> = {
+export type OnSubscribeResultEventPayload<TInputContext, TOutputContext> = {
   /**
    * The execution arguments.
    */
-  args: TypedExecutionArgs<ContextType>;
+  args: TypedExecutionArgs<TInputContext & TOutputContext>;
```

Comment:
Are you sure this is correct? TOutputContext should overwrite context values from TInputContext

TInputContext
```
{
  a: 1,
  b: 2.
}
```

TOutputContext
```
{
  a: 3.
  c: 1,
}
```

Result:

```
{
  a: 3,
  b: 2.
  c: 1,
}
```

Currently, it would result in

```
{
  a: 1 | 3,
  b: 2.
  c: 1,
}
```